backend/mongoose/functions/user_functions.js

```diff
@@ -119,6 +119,31 @@ const getUserByUsername = async (un) => {
   }
 };
 
+const getUserById = async (id) => {
+  try {
+    const user = await User.findById(id).select('-password').select('-salt');
+    if (user === null) {
+      return {
+        success: false,
+        error: 'no match found',
+        code: 404
+      };
+    } else {
+      return {
+        success: true,
+        code: 200,
+        user: user
+      };
+    }
+  } catch (error) {
+    return {
+      success: false,
+      error: error.message,
+      code: 500
+    };
```

Comment:
We're also catching errors in the `user` service itself, right? I removed this extra try-catch from my service functions and they seem to handle errors alright.

Reply:
Yes, that's correct.  I included it as kinda a failsafe, there was a point or two where it was useful, but you're right.